File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/util/CookieUtils.java
package org.javamaster.invocationlab.admin.util;

import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletRequest;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * @author yudong
 */
@SuppressWarnings("DataFlowIssue")
public class CookieUtils {

    public static String getCookieValue(String name) {
        ServletRequestAttributes requestAttributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
        HttpServletRequest request = requestAttributes.getRequest();
        Cookie[] cookies = request.getCookies();
        if (cookies == null) {
            return "";
        }
        List<Cookie> list = Arrays.stream(cookies)
                .filter(cookie -> name.equals(cookie.getName()))
                .collect(Collectors.toList());
        if (list.isEmpty()) {
            return "";
        }
        return list.get(0).getValue();
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/util/CookieUtils.kt
package org.javamaster.invocationlab.admin.util

import jakarta.servlet.http.Cookie
import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes
import java.util.*

/**
 * @author yudong
 */
object CookieUtils {
    @JvmStatic
    fun getCookieValue(name: String): String {
        val requestAttributes = RequestContextHolder.getRequestAttributes() as ServletRequestAttributes?
        val request = requestAttributes!!.request
        val cookies = request.cookies ?: return ""
        val list = Arrays.stream(cookies)
            .filter { cookie: Cookie -> name == cookie.name }
            .toList()
        if (list.isEmpty()) {
            return ""
        }
        return list.first().value
    }
}
